build.gradle

```diff
@@ -295,6 +295,8 @@ dependencies {
     testCompile group: 'net.serenity-bdd', name: 'serenity-spring', version: versions.serenity
     testCompile "com.github.tomakehurst:wiremock-jre8:2.23.2"
 
+    testCompile 'pl.pragmatists:JUnitParams:1.1.1'
```

Comment:
@raghera can you take a look at this library? I like the idea, but I am not sure if it is still supported (the last commit to master was in 2018).
https://github.com/Pragmatists/JUnitParams

Reply:
JUnit offers its own way to parametrise tests: https://github.com/junit-team/junit4/wiki/Parameterized-tests. It looks similar but my understanding is that is not that straightforward. 

In practice, it is really useful and in line with the OO best practice since allows you to reuse code and use the same test to cover different scenarios as you see. 

in SSCS we use it all over the place and it has not created issues at all if that helps. Regarding your concerns, in the long run, I guess we could always re-write the tests, or use the JUnit approach. 

Anyway, I just opened an issue to address your concerns: https://github.com/Pragmatists/JUnitParams/issues/169

If you agree, guys, we can create a tech-debt task and put it on our board and I can pick it up in between tickets to re-write it using the junit approach to make it safer :)

Reply:
I think it's nice to be able to add params to tests (other frameworks like TestNG have had it for ages).  I also think the Pragmatists lib is much better than the JUnit4 version although the support concern is legitimate and thanks for raising with the team David!  What mitigates that a little is that in future we should maybe move to JUnit5 which has this feature out of the box: https://www.baeldung.com/parameterized-tests-junit-5

Reply:
totally agree with you, Ravi! the parameters functionality in Junit4 has important limitations. Junit5 looks much better and once we move on that we should use that approach, yes.